lib/puppet/type/archive.rb

```diff
@@ -150,6 +150,16 @@ def should_to_s(value)
     desc 'extract command group (using this option will configure the archive file permisison to 0644 so the user can read the file).'
   end
 
+  newparam(:proxy_type) do
+    desc 'proxy type (none|ftp|http|https)'
+    newvalues(:none, :ftp, :http, :https)
+    defaultto(:none)
```

Comment:
We can defaultto a sensible value based on proxy_server and let user override. Let me find you an example.